src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const key in obj.extraData) {
+          stateCopy[key] = obj.extraData[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'removeProperties':
+        for (const key of obj.keysToRemove) {
+          delete stateCopy[key];
+        }
+        stateHistory.push({ ...stateCopy });
+        break;
+
+      case 'clear':
+        for (const key in stateCopy) {
+          delete stateCopy[key];
+        }
+        stateHistory.push({ ...stateCopy });
```

Comment:
To clear the `stateCopy`, instead of deleting all properties one by one, you should simply assign a new empty object to a new variable to represent the cleared state.

Reply:
fixed